test/e2e/marketplace_test.go

```diff
@@ -0,0 +1,186 @@
+			RegistryNamespace: "operators",
+		},
+	}
+
+	catalogSourceConfigName := "opsrc-global-operators"
+	configMapName := "csc-cm-opsrc-global-operators"
+	catalogSourceName := "csc-cs-opsrc-global-operators"
+
+	// Create the operatorsource to download the manifests.
+	err = f.Client.Create(
+		context.TODO(),
+		testOperatorSource,
+		&framework.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       cleanupTimeout,
+			RetryInterval: cleanupRetryInterval,
+		})
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig.
+	testCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, testCatalogSourceConfig, namespace, catalogSourceConfigName)
+	if err != nil {
+		return err
+	}
+
+	// Check for the config map created from the catalogsourceconfig.
+	err = WaitForResult(t, f, &corev1.ConfigMap{}, namespace, configMapName)
+	if err != nil {
+		return err
+	}
+
+	// Then check for the catalog source.
+	err = WaitForResult(t, f, &olm.CatalogSource{}, namespace, catalogSourceName)
+	if err != nil {
+		return err
+	}
+
+	// Now lets update the catalogsourceconfig and see that the configmap and catalog source are updated
+	// First set the target namespace field for the catalogsourceconfig to a new namespace
+	testCatalogSourceConfig.Spec.TargetNamespace = testNamespace
+	err = f.Client.Update(context.TODO(), testCatalogSourceConfig)
+	if err != nil {
+		return err
+	}
+
+	// Check for the config map created in the new namespace
+	err = WaitForResult(t, f, &corev1.ConfigMap{}, testNamespace, configMapName)
+	if err != nil {
+		return err
+	}
+
+	// Then check for the catalog source in the new namespace
+	err = WaitForResult(t, f, &olm.CatalogSource{}, testNamespace, catalogSourceName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func MarketplaceCluster(t *testing.T) {
+	t.Parallel()
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+	err := ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	if err != nil {
+		t.Fatalf("failed to initialize cluster resources: %v", err)
+	}
+	t.Log("Initialized cluster resources")
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	// get global framework variables
+	f := framework.Global
+	// wait for marketplace-operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "marketplace-operator", 1, retryInterval, timeout)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = defaultCreateTest(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func WaitForResult(t *testing.T, f *framework.Framework, result runtime.Object, namespace, name string) error {
+	namespacedName := types.NamespacedName{Name: name, Namespace: namespace}
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		err = f.Client.Get(context.TODO(), namespacedName, result)
+		if apierrors.IsNotFound(err) {
+			if err != nil {
```

Comment:
Is this check necessary? The two return statements following are equivalent, so is this just to control when to print logs?

Reply:
Yeah, how can `err` be `nil` at this point?

Reply:
I have misspoken, it will not panic.